File: i2f-springboot/i2f-springboot-secure-starter/src/main/java/i2f/springboot/swl/std/ISwlSymmetricEncryptor.java
package i2f.springboot.swl.std;

/**
 * @author Ice2Faith
 * @date 2024/7/9 18:41
 * @desc
 */
public interface ISwlSymmetricEncryptor {
    String generateKey();

    String getKey();

    void setKey(String key);

    String encrypt(String data);

    String decrypt(String data);
}

File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-swl/core/std/ISwlSymmetricEncryptor.js
/**
 * @return {ISwlSymmetricEncryptor}
 * @interface ISwlSymmetricEncryptor
 */
function ISwlSymmetricEncryptor() {
    return {
        /**
         * @return {String}
         */
        generateKey() {

        },
        /**
         * @return {String}
         */
        getKey() {

        },
        /**
         *
         * @param key {String}
         * @return {void}
         */
        setKey(key) {

        },
        /**
         *
         * @param data {String}
         * @return {String}
         */
        encrypt(data) {

        },
        /**
         *
         * @param data {String}
         * @return {String}
         */
        decrypt(data) {

        }
    }
}

export default ISwlSymmetricEncryptor
